src/common/types.hpp

```diff
@@ -314,6 +322,30 @@ class Ip6Prefix
      */
     Ip6Prefix(void) { Clear(); }
 
+    /**
+     * Constructor with an Ip6 address string and prefix length.
+     *
+     * @param[in] aIp6AddrStr The IPv6 address string.
+     * @param[in] aLength     The prefix length.
+     *
+     */
+    Ip6Prefix(const char *aIp6AddrStr, uint8_t aLength)
+        : mPrefix(aIp6AddrStr)
+        , mLength(aLength)
+    {
+    }
+
+    /**
+     *
+     * This method overloads `==` operator for comparing two Ip6Prefix objects by comparing their prefix and length.
+     *
+     * @param[in] aOther The Ip6Prefix object to compare with.
+     *
+     * @returns True if the prefix and length of the two objects are equal, false otherwise.
+     *
+     */
+    bool operator==(const Ip6Prefix &aOther) const { return mPrefix == aOther.mPrefix && mLength == aOther.mLength; }
```

Comment:
For mPrefix == aOther.mPrefix, do you only compare the pointer, or you want to compare the content of the char?
For mLength == aOther.mLength, this may be moved before mPrefix compare since int compare could be faster than string content comparison.

Reply:
1. I expect it calls the `operator==` of class `Ip6Prefix` in line# 347
2. good point! updated